libbeat/outputs/kafka/config.go

```diff
@@ -169,6 +175,20 @@ func (c *kafkaConfig) Validate() error {
 			return fmt.Errorf("compression_level must be between 0 and 9")
 		}
 	}
+
+	if c.Topic == "" && len(c.Topics) == 0 {
```

Comment:
Would it be possible to attempt to evaluate `c.Topic` as a format string and log a warning if we think it is one?

That would help people figure out why their topic routing is failing. I don't think the `%{` syntax is a valid topic name so the only way we'd get that is if someone is trying to use the Beat topic syntax with agent.

Reply:
The bits of code that used to convert the `%{foo}` notation is deep within some helper from the outputs (`outil.BuildSelectorFromConfig`) and it will return some errors that might not be what we want.

I could either search for `%{` in the topic name and return an error or use a regexp validation with the characters Lee pointed. I prefer the latter and returning an error message stating the valid characters.

Reply:
Agreed just use the character set Lee linked and link back to that file in our code.